src/config.rs

```diff
@@ -221,25 +258,81 @@ impl OrchestratorConfig {
     }
 
     fn validate(&self) -> Result<(), Error> {
+        // Detectors are configured
         if self.detectors.is_empty() {
-            Err(Error::NoDetectorsConfigured)
-        } else {
-            for (detector_id, detector) in &self.detectors {
-                // Chunker is valid
-                let valid_chunker = detector.chunker_id == DEFAULT_MODEL_ID
-                    || self
-                        .chunkers
-                        .as_ref()
-                        .is_some_and(|chunkers| chunkers.contains_key(&detector.chunker_id));
-                if !valid_chunker {
-                    return Err(Error::DetectorChunkerNotFound {
-                        detector_id: detector_id.clone(),
-                        chunker_id: detector.chunker_id.clone(),
-                    });
+            return Err(Error::NoDetectorsConfigured);
+        }
+
+        // Detector configs are valid
+        for (detector_id, detector) in &self.detectors {
+            // Hostname is valid
+            if !is_valid_hostname(&detector.service.hostname) {
+                return Err(Error::InvalidHostname(format!(
+                    "detector `{detector_id}` has an invalid hostname"
+                )));
+            }
+            // Chunker is valid
+            let valid_chunker = detector.chunker_id == DEFAULT_MODEL_ID
+                || self
+                    .chunkers
+                    .as_ref()
+                    .is_some_and(|chunkers| chunkers.contains_key(&detector.chunker_id));
+            if !valid_chunker {
+                return Err(Error::DetectorChunkerNotFound {
+                    detector_id: detector_id.clone(),
+                    chunker_id: detector.chunker_id.clone(),
+                });
+            }
+        }
+
+        // Chunker config is valid
+        if let Some(chunkers) = &self.chunkers {
+            for (chunker_id, chunker) in chunkers {
+                // Hostname is valid
+                if !is_valid_hostname(&chunker.service.hostname) {
+                    return Err(Error::InvalidHostname(format!(
+                        "chunker `{chunker_id}` has an invalid hostname"
+                    )));
                 }
             }
-            Ok(())
         }
+
+        // Generation config is valid
+        if let Some(generation) = &self.generation {
+            // Provider is valid
+            if !matches!(
+                generation.provider,
```

Comment:
Q: if we separate out `chat_provider` from `generation_provider`, then it would simplify this sort of validation right?

Any reasons for putting these providers together in same `enum` ?

Reply:
Yes, I did this as I anticipate we will adopt the OpenAI API in the future instead of continuing to maintain the TGIS compatibility layer.

Reply:
I just pushed a change to drop `provider` from `ChatGenerationConfig` (since it is always openai for this) and `GenerationProvider::OpenAi` variant. Also dropped provider validation rules as they aren't needed now.